spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerOperations.java

```diff
@@ -77,10 +78,27 @@
 	 */
 	<T> List<T> read(Class<T> entityClass, KeySet keys);
 
+	/**
+	 * Executes a given query string with tags and parameters and applies a given function
+	 * to each row of the result.
+	 * @param rowFunc the function to apply to each row of the result.
+	 * @param entityClass the underlying domain type whose properties are used in sorts in
+	 * the result.
+	 * @param sql the query string to run.
+	 * @param tags the tags that appear in the query.
+	 * @param params the value to bind to those tags.
+	 * @param options the options with which to run this query.
+	 * @param <T> the type to transform each row into.
+	 * @return a list of the rows each transformed with the given function.
+	 */
+	<A, T> List<A> query(Function<Struct, A> rowFunc, Class<T> entityClass, String sql,
```

Comment:
See if you can get rid of the `entityClass` as well as `SpannerQueryOptions`.
We should probably assume that `sql` is already constructed with the options that can go into it.